lib/practice_exercises.rb

```diff
@@ -1,13 +1,47 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  list = []
+
+  c_string = string.gsub(/[^a-zA-Z]/, "").downcase
+  index = c_string.length - 1
+
+  c_string.each_char do |letter|
+    if letter == c_string[index]
+      index -= 1
+    else 
+      list<< 1
+      index -= 1
+    end
+    if index == (c_string.length)/2
+      if 
+        list == [] 
+        return true
+      else 
+        return false
+      end
+    end
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍